### PR TITLE
friends: ability to get Friend Code

### PR DESCRIPTION
**Pull request by TurtleP**
This pull request adds the ability to open the friend:* services and get the friend code, along with other items.

**Hardware Test** 
image

**Tested Source Code**
[Friend-Code.zip](https://github.com/switchbrew/libnx/files/6272279/Friend-Code.zip)

### CODE REVIEW

File: nx/include/switch/services/friends.h [hunk 1]
```diff
@@ -32,6 +40,36 @@ typedef struct {
...
+} FriendUserSetting;
+
+/// Initialize friends
+Result friendsInitialize(FriendServiceType service_type);
+
+/// Exit friends
+void friendsExit(void);
+
+/// Gets the Service object for the friends service session.
+Service * friendsGetServiceSession(void);
```
Comment (yellows8) [inline nx/include/switch/services/friends.h:61]: `Service*`

File: nx/include/switch/services/friends.h [hunk 2]
```diff
@@ -32,6 +40,36 @@ typedef struct {
...
+
+/// Gets the Service object for the actual IFriendsService service session.
+Service* friendsGetServiceSession_IFriendsService(void);
+
+/**
+ * @brief Gets the \ref FriendUserSetting details
+ * @param[in] uid \ref User AccountUid.
+ * @param[out] user_setting \ref FriendUserSetting
+ */
+Result friendsGetUserSetting(AccountUid uid, FriendUserSetting * user_setting);
```
Comment (yellows8) [inline nx/include/switch/services/friends.h:71]: `*user_setting`

**Comment by TurtleP** [comment]
Something interesting to note, I guess: to actually *get* the friend code, you have to open the service with Viewer or higher permissions. User does not seem to allow fetching it.

File: nx/source/services/friends.c [hunk 3]
```diff
@@ -1,6 +1,65 @@
...
+Service* friendsGetServiceSession(void) {
+    return &g_friendSrv;
+}
+
+Service* friendsGetServiceSession_IFriendsService(void)
+{
+    return &g_friendsIFriendService;
+}
+
+Result friendsGetUserSetting(AccountUid uid, FriendUserSetting * user_setting) {
```
Comment (yellows8) [inline nx/source/services/friends.c:56]: `*user_setting`

**Comment by TurtleP** [comment]
Updated the source code test to reflect the new changes that were pushed

**Review by yellows8 — APPROVED** [review]

File: nx/include/switch/services/friends.h [hunk 4]
```diff
@@ -11,6 +11,14 @@
 #include "../services/acc.h"
 #include "../sf/service.h"
 
+typedef enum {
+    FriendsServiceType_User          = 0, ///< Initializes friend:u
+    FriendsServiceType_Viewer        = 1, ///< Initializes friend:v
+    FriendsServiceType_Manager       = 2, ///< Initializes friend:m
+    FriendsServiceType_System        = 3, ///< Initializes friend:s
+    FriendsServiceType_Administrator = 4  ///< Initializes friend:a
```
Comment (fincs) [inline nx/include/switch/services/friends.h:19]: Missing trailing comma

**Review by fincs — APPROVED** [review]

**Pull request merged**